### PR TITLE
fix(Table): remove redundant props from DOM attributes

### LINKED ISSUE #1449: Bug(withTableSettings): redundant props in withSettingsTable which shouldn't be in DOM attributes
`isRequired` and `isSelected` props should me omitted from [`mapItemDataToProps`](https://github.com/gravity-ui/uikit/blob/fdbb0525bf9991baa68ff0d3786050cac480f78f/src/components/Table/hoc/withTableSettings/TableColumnSetup/TableColumnSetup.tsx#L285) which comes to `TreeSelect.TreeList.renderItem`, which exposes all `item` data to the `renderItem` as `renderState.props`.

<details><summary>Console errors</summary>
<p>

```
Warning: React does not recognize the `isSelected` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `isselected` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    at div
    at Box (http://localhost:7007/src_components_layout_Box_Box_tsx.iframe.bundle.js:42:3)
    at Flex (http://localhost:7007/src_components_layout_Flex_Flex_tsx.iframe.bundle.js:72:5)
    at _c2 (http://localhost:7007/src_components_useList_components_ListItemView_ListItemView_tsx-src_components_useList_hooks_-acae8c.iframe.bundle.js:69:3)
    at Draggable (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:8690:58)
    at ConnectFunction (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:9931:68)
    at PrivateDraggable (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:9042:26)
    at PublicDraggable (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:9052:32)
    at div
    at Droppable (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:9064:69)
    at ConnectFunction (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:9931:68)
    at Provider (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:9643:20)
    at App (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:7661:25)
    at ErrorBoundary (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:924:35)
    at DragDropContext (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:7790:19)
    at div
    at Box (http://localhost:7007/src_components_layout_Box_Box_tsx.iframe.bundle.js:42:3)
    at Flex (http://localhost:7007/src_components_layout_Flex_Flex_tsx.iframe.bundle.js:72:5)
    at ListContainerView (http://localhost:7007/src_components_useList_components_ListContainerView_ListContainerView_tsx-src_components_useL-3f175c.iframe.bundle.js:31:3)
    at TreeList (http://localhost:7007/src_components_TreeSelect_TreeSelect_tsx.iframe.bundle.js:37:3)
    at div
    at FocusTrap (http://localhost:7007/src_components_Portal_Portal_tsx-src_components_utils_FocusTrap_tsx-src_components_utils_laye-678600.iframe.bundle.js:123:3)
    at div
    at Transition (http://localhost:7007/vendors-node_modules_react-transition-group_esm_CSSTransition_js.iframe.bundle.js:685:30)
    at CSSTransition (http://localhost:7007/vendors-node_modules_react-transition-group_esm_CSSTransition_js.iframe.bundle.js:235:35)
    at Portal (http://localhost:7007/src_components_Portal_Portal_tsx-src_components_utils_FocusTrap_tsx-src_components_utils_laye-678600.iframe.bundle.js:29:3)
    at Popup (http://localhost:7007/src_components_Popup_Popup_tsx.iframe.bundle.js:49:3)
    at _c (http://localhost:7007/src_components_Select_components_SelectControl_SelectControl_tsx-src_components_Select_compon-6c14fc.iframe.bundle.js:366:3)
    at div
    at Box (http://localhost:7007/src_components_layout_Box_Box_tsx.iframe.bundle.js:42:3)
    at Flex (http://localhost:7007/src_components_layout_Flex_Flex_tsx.iframe.bundle.js:72:5)
    at TreeSelect (http://localhost:7007/src_components_TreeSelect_TreeSelect_tsx.iframe.bundle.js:331:3)
    at TableColumnSetup (http://localhost:7007/components-Table-__stories__-Table-stories.iframe.bundle.js:1613:5)
    at div
    at th
    at tr
    at thead
    at table
    at div
    at Table (http://localhost:7007/src_components_Table_Table_tsx.iframe.bundle.js:60:1)
    at TableWithSettings (http://localhost:7007/components-Table-__stories__-Table-stories.iframe.bundle.js:2004:7)
    at MobileProvider (http://localhost:7007/main.iframe.bundle.js:819:3)
    at ThemeProvider (http://localhost:7007/main.iframe.bundle.js:1068:10)
    at unboundStoryFn (http://localhost:7007/sb-preview/runtime.js:41:3662)
    at ErrorBoundary (http://localhost:7007/vendors-node_modules_bem-react_classname_index_js-node_modules_pmmmwh_react-refresh-webpack-p-a79f08.iframe.bundle.js:4266:439)
    at WithCallback (http://localhost:7007/vendors-node_modules_bem-react_classname_index_js-node_modules_pmmmwh_react-refresh-webpack-p-a79f08.iframe.bundle.js:4238:34)
```
```
Warning: React does not recognize the `isRequired` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `isrequired` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    at div
    at Box (http://localhost:7007/src_components_layout_Box_Box_tsx.iframe.bundle.js:42:3)
    at Flex (http://localhost:7007/src_components_layout_Flex_Flex_tsx.iframe.bundle.js:72:5)
    at _c2 (http://localhost:7007/src_components_useList_components_ListItemView_ListItemView_tsx-src_components_useList_hooks_-acae8c.iframe.bundle.js:69:3)
    at Draggable (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:8690:58)
    at ConnectFunction (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:9931:68)
    at PrivateDraggable (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:9042:26)
    at PublicDraggable (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:9052:32)
    at div
    at Droppable (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:9064:69)
    at ConnectFunction (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:9931:68)
    at Provider (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:9643:20)
    at App (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:7661:25)
    at ErrorBoundary (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:924:35)
    at DragDropContext (http://localhost:7007/vendors-node_modules_gravity-ui_icons_esm_Grip_js-node_modules_gravity-ui_icons_esm_TriangleE-693099.iframe.bundle.js:7790:19)
    at div
    at Box (http://localhost:7007/src_components_layout_Box_Box_tsx.iframe.bundle.js:42:3)
    at Flex (http://localhost:7007/src_components_layout_Flex_Flex_tsx.iframe.bundle.js:72:5)
    at ListContainerView (http://localhost:7007/src_components_useList_components_ListContainerView_ListContainerView_tsx-src_components_useL-3f175c.iframe.bundle.js:31:3)
    at TreeList (http://localhost:7007/src_components_TreeSelect_TreeSelect_tsx.iframe.bundle.js:37:3)
    at div
    at FocusTrap (http://localhost:7007/src_components_Portal_Portal_tsx-src_components_utils_FocusTrap_tsx-src_components_utils_laye-678600.iframe.bundle.js:123:3)
    at div
    at Transition (http://localhost:7007/vendors-node_modules_react-transition-group_esm_CSSTransition_js.iframe.bundle.js:685:30)
    at CSSTransition (http://localhost:7007/vendors-node_modules_react-transition-group_esm_CSSTransition_js.iframe.bundle.js:235:35)
    at Portal (http://localhost:7007/src_components_Portal_Portal_tsx-src_components_utils_FocusTrap_tsx-src_components_utils_laye-678600.iframe.bundle.js:29:3)
    at Popup (http://localhost:7007/src_components_Popup_Popup_tsx.iframe.bundle.js:49:3)
    at _c (http://localhost:7007/src_components_Select_components_SelectControl_SelectControl_tsx-src_components_Select_compon-6c14fc.iframe.bundle.js:366:3)
    at div
    at Box (http://localhost:7007/src_components_layout_Box_Box_tsx.iframe.bundle.js:42:3)
    at Flex (http://localhost:7007/src_components_layout_Flex_Flex_tsx.iframe.bundle.js:72:5)
    at TreeSelect (http://localhost:7007/src_components_TreeSelect_TreeSelect_tsx.iframe.bundle.js:331:3)
    at TableColumnSetup (http://localhost:7007/components-Table-__stories__-Table-stories.iframe.bundle.js:1613:5)
    at div
    at th
    at tr
    at thead
    at table
    at div
    at Table (http://localhost:7007/src_components_Table_Table_tsx.iframe.bundle.js:60:1)
    at TableWithSettings (http://localhost:7007/components-Table-__stories__-Table-stories.iframe.bundle.js:2004:7)
    at MobileProvider (http://localhost:7007/main.iframe.bundle.js:819:3)
    at ThemeProvider (http://localhost:7007/main.iframe.bundle.js:1068:10)
    at unboundStoryFn (http://localhost:7007/sb-preview/runtime.js:41:3662)
    at ErrorBoundary (http://localhost:7007/vendors-node_modules_bem-react_classname_index_js-node_modules_pmmmwh_react-refresh-webpack-p-a79f08.iframe.bundle.js:4266:439)
    at WithCallback (http://localhost:7007/vendors-node_modules_bem-react_classname_index_js-node_modules_pmmmwh_react-refresh-webpack-p-a79f08.iframe.bundle.js:4238:34)
```

</p>
</details>

### PR DESCRIPTION
**Pull request by teleginzhenya**
resolves #1449 

cc @IsaevAlexandr

### CODE REVIEW

File: src/components/Table/hoc/withTableSettings/TableColumnSetup/TableColumnSetup.tsx [hunk 1]
```diff
@@ -24,7 +24,10 @@ import i18n from './i18n';
 
 import './TableColumnSetup.scss';
 
-function identity<T>(value: T): T {
+function identity(value: Item): ItemRenderProps {
+    delete value.isSelected;
+    delete value.isRequired;
```
Comment (amje) [inline src/components/Table/hoc/withTableSettings/TableColumnSetup/TableColumnSetup.tsx:29]: Mutating parameter is bad, can we resolve this another way?

Comment (teleginzhenya) [reply src/components/Table/hoc/withTableSettings/TableColumnSetup/TableColumnSetup.tsx]: Sure, 7f624b2a1726f0f2d24d5f3d290c489d62537fc3

File: src/components/Table/hoc/withTableSettings/TableColumnSetup/TableColumnSetup.tsx [hunk 2]
```diff
@@ -171,6 +173,8 @@ type Item = TableColumnSetupItem &
         isDragDisabled?: boolean;
     };
 
+type ItemRenderProps = Omit<Item, 'isSelected' | 'isRequired'>;
```
Comment (amje) [inline src/components/Table/hoc/withTableSettings/TableColumnSetup/TableColumnSetup.tsx:176]: This type not used anywhere and not exported. Why it needed?

Comment (teleginzhenya) [reply src/components/Table/hoc/withTableSettings/TableColumnSetup/TableColumnSetup.tsx:176]: It is used in line 27 https://github.com/gravity-ui/uikit/pull/1450/files/d80cfcb5ace703089ef64e442810607f699cfda1#diff-6baaa789dea19c3913ee44fcc6ebac7f3c704b67dd7007e8e8685bd4639928c9R27

**Review by amje — APPROVED** [review]

**Comment by teleginzhenya** [comment]
@Raubzeug hey! can you please take a look?

**Review by Raubzeug — APPROVED** [review]

**Comment by teleginzhenya** [comment]
This is not working without change List components, because `isSelected`/`isRequired` comes from `TableColumnSetupItem` as well

**Pull request closed without merging**